### PR TITLE
Script for preparing calibration and test sets 

### PR DESCRIPTION
**Pull request by fatihcakirs**
Related issue is #108. Previous related (unmerged) PR is https://github.com/mlcommons/mobile_app_open/pull/123. 

The script prepares and outputs the test sets to output/mlperf_datasets folder, which can directly be copied to a device for accuracy evaluation (excludes any ground truth files). I've used the processing scripts provided by MLPerf.

Calibration sets are in output/calibration_set folder. 

Calibration images (except for SQuAD) are not processed. Submitters may need to further preprocess calibration data for compatibility with their SDKs.

### CODE REVIEW

File: datasets/Makefile [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+USERID = $(shell id -u)
+GROUPID = $(shell id -g)
+
+# Ade20k calibration vars
+A_SRC_DIR := ${BUILDDIR}/downloads/ADEChallengeData2016/images/training
+A_FILES = $(shell cat ${BUILDDIR}/util/ade20k/ade20k_calibration_files.txt)
+A_SRC = $(addprefix $(A_SRC_DIR)/,$(A_FILES))
+A_DEST_DIR := ${BUILDDIR}/output/calibration_set/ade20k/images
+
+A_TARGS := $(patsubst $(A_SRC_DIR)/%.jpg,$(A_DEST_DIR)/%.jpg,$(A_SRC))
```
Comment (mcharleb) [inline datasets/Makefile:34]: Needs to be A_TARGS := $(patsubst $(A_SRC_DIR)/%.jpg,$(A_DEST_DIR)/training/%.jpg,$(A_SRC))

**Comment by mcharleb** [comment]
We will need to work around web proxy as I had done using chunk downloads:

Fetching COCO val2014.zip
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  8 6337M    8  536M    0     0   351k      0  5:07:17  0:26:00  4:41:17  120k
 12 6337M   12  811M    0     0   353k      0  5:06:00  0:39:10  4:26:50  105k
curl: (18) transfer closed with 5794447890 bytes remaining to read

**Review by mcharleb — CHANGES_REQUESTED** [review]

**Comment by mcharleb** [comment]
@d-uzlov  Not sure why the code format check is failing. In the logs I see the following, but running clang-format on these files makes no changes:

/var/lib/jenkins/workspace/mobile_app_open/PullRequest-CodeFormattingCheck:/home/mlperf/mobile_app -w /home/mlperf/mobile_app -u 109:114 mlcommons/mlperf_mobile:1.0 android/formatters/run-format-code.sh
Updated/new C++ files: android/cpp/backends/external.cc android/cpp/utils.cc flutter/cpp/backend.h flutter/cpp/backends/external.cc flutter/cpp/backends/external.h flutter/cpp/binary/main.cc flutter/cpp/c/backend_c.h flutter/cpp/c/type.h flutter/cpp/dataset.h flutter/cpp/datasets/ade20k.cc flutter/cpp/datasets/ade20k.h flutter/cpp/datasets/allocator.cc flutter/cpp/datasets/allocator.h flutter/cpp/datasets/coco.cc flutter/cpp/datasets/coco.h flutter/cpp/datasets/imagenet.cc flutter/cpp/datasets/imagenet.h flutter/cpp/datasets/squad.cc flutter/cpp/datasets/squad.h flutter/cpp/datasets/squad_utils/common.h flutter/cpp/datasets/squad_utils/tfrecord_reader.h flutter/cpp/datasets/squad_utils/types.h flutter/cpp/flutter/dart_backend_match.cc flutter/cpp/flutter/dart_mlperf_config.cc flutter/cpp/flutter/main.cc flutter/cpp/flutter/main.h flutter/cpp/mlperf_driver.cc flutter/cpp/mlperf_driver.h flutter/cpp/proto/test.cc flutter/cpp/utils.cc flutter/cpp/utils.h flutter/cpp/utils_test.cc
No such file or directory
No such file or directory
No such file or directory
No such file or directory
No such file or directory
No such file or directory
No such file or directory
No such file or directory
No such file or directory
No such file or directory
No such file or directory
No such file or directory
No such file or directory
No such file or directory
No such file or directory
No such file or directory
No such file or directory
No such file or directory
No such file or directory
No such file or directory
No such file or directory
No such file or directory
No such file or directory
No such file or directory
No such file or directory
No such file or directory
Build step 'Run with timeout' marked build as failure
Archiving artifacts
Setting status of c5793469e429d843d42faa781c37befd02f0239c to FAILURE with url https://mlperf.xored.com/job/mobile_app_open/job/PullRequest-CodeFormattingCheck/127/ and message: 'Failed
 '
Using context: Code formatting
Finished: FAILURE

**Review by mcharleb — APPROVED** [review]

**Comment by d-uzlov** [comment]
Running the formatting script in prep-datasets branch doesn't format anything, but if I merge it into the master branch I do get the exact same error.
Seems like our formatting script tries to format files deleted in the previous PRs, and fails. We should probably fix it later.

@fatihcakirs can you rebase you branch to the last commit in the master branch?

**Comment by mcharleb** [comment]
@fatihcakirs and @d-uzlov OK if I merge this?

**Comment by d-uzlov** [comment]
@mcharleb can we add some documentation to the `/datasets` folder?
If we already have one elsewhere, can you at least create a `/datasets/readme` file with a link to the main documentation?

Oh, and  I guess you forgot to update the Folder Structure section of the repo readme.

**Comment by fatihcakirs** [comment]
Thank you @d-uzlov and @mcharleb , I've added a brief description to the `Folder Structure` section in the repo readme. This should suffice in my opinion, the make file is itself self-explanatory. Unless you need a more detailed readme file in the `datasets` folder, I'm fine with the merge.

**Review by d-uzlov — APPROVED** [review]

**Pull request merged**